components/kyma-environment-broker/internal/error/error.go

```diff
@@ -0,0 +1,38 @@
+package error
+
+import "fmt"
+
+type TemporaryError struct {
+	message string
+}
+
+func NewTemporaryError(message string, err error) *TemporaryError {
```

Comment:
Personally I do not like functions where optional parameters are not visible and for some args I can pass nil.
```go
kebError.NewTemporaryError("response status from director is nil", nil)
kebError.NewTemporaryError("while requesting to director client", err)
```

IMO better will be used functional option to construct such error, or introduce two methods:

```go
// creates a new tmp error
func NewTemporaryError(msg string, args ...interface{}) *TemporaryError {
	return &TemporaryError{message: fmt.Sprintf(msg, args..)}
}

// wrap existing one as tmp error
func AsTemporaryError(err error, context string, args ...interface{})  *TemporaryError {
	errCtx := fmt.Sprintf(context, args...)
	msg := fmt.Sprintf("%s: %s", errCtx, err.Error())

	return &TemporaryError{message: msg}
}
```
Usage
```go
kebError.NewTemporaryError("response status from director is nil")

kebError.AsTemporaryError(err, "while requesting to director client")

kebError.AsTemporaryError(err, "while deleting object with id: %s", obj.ID)
```